v3data/__init__.py

```diff
@@ -29,8 +29,20 @@ async def query(self, query: str, variables=None) -> dict:
             params = {"query": query, "variables": variables}
         else:
             params = {"query": query}
+        # TODO: error handling -> connection, result and others   httpcore.RemoteProtocolError
+        #                                                         ssl.SSLError:   [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure
+        #
         response = await async_client.post(self._url, json=params)
-        return response.json()
+        if response.status_code != 204 and response.headers[
```

Comment:
This seems a little convoluted.  Do you have a sample of a response that we want to catch errors for?

Reply:
I have sporadic ssl Errors and   while filling database fee yields and impermanent data. I will try to reproduce and come back to u with examples... From my experience, thegraph does not always respond an object nor common error. ... but it may be my internet connection ..

Reply:
I did a bit of simple testing just now.
if the query request is successful, the status code always = 200, regardless of whether the gql query is valid or not.
If there is an error with the gql query, the error response will look something like this:
```
{
  "errors": [
    {
      "locations": [
        {
          "line": 4,
          "column": 5
        }
      ],
      "message": "Type `MasterChef` has no field `a`"
    }
  ]
}
```

Maybe we can do something along these lines which would be easier to reason with?

```
if response.status_code == "200":
    if parsed_response := response.json():
        return parsed_response
    else:
        # handle bad gql query
else:
    # handle bad status code
    # Can expand this to handle specific codes once we have specific examples
```